android/java/org/chromium/chrome/browser/crypto_wallet/fragments/MarketFragment.java

```diff
@@ -0,0 +1,185 @@
+        mShimmerViewContainer = mRootView.findViewById(R.id.shimmer_view_container);
+        mSingleItemShimmer = mRootView.findViewById(R.id.market_coin_item);
+        mShimmerCardView = mRootView.findViewById(R.id.card_view_market_placeholder);
+        mShimmerListContainer = mRootView.findViewById(R.id.shimmer_list);
+        mMarketListCardView = mRootView.findViewById(R.id.market_list_card_view_view_container);
+
+        mCoinMarkets = mRootView.findViewById(R.id.coin_markets_recycler_view);
+        mCoinMarkets.addItemDecoration(
+                new DividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL));
+        mCoinMarkets.setAdapter(mMarketCoinAdapter);
+        mCoinMarkets.setLayoutManager(new LinearLayoutManager(getActivity()));
+        mMarketModel.getCoinMarkets(this);
+
+        final int visibleShimmerItems =
+                getVisibleShimmerItems(mSingleItemShimmer, mShimmerCardView);
+        for (int i = 1; i < visibleShimmerItems; i++) {
+            LayoutInflater.from(requireContext())
+                    .inflate(R.layout.brave_wallet_divider, mShimmerListContainer, true);
+            LayoutInflater.from(requireContext())
+                    .inflate(R.layout.market_coin_shimmer_item, mShimmerListContainer, true);
+        }
+
+        return mRootView;
+    }
+
+    private int getVisibleShimmerItems(
+            final ViewGroup mSingleItemShimmer, final CardView shimmerCardView) {
+        int widthMeasureSpec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED);
+        int heightMeasureSpec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED);
+
+        mSingleItemShimmer.measure(widthMeasureSpec, heightMeasureSpec);
+        int itemHeight = mSingleItemShimmer.getMeasuredHeight();
+        int totalHeight = AndroidUtils.getScreenHeight();
+
+        int totalHeightNoPadding =
+                totalHeight - shimmerCardView.getPaddingTop() - shimmerCardView.getPaddingBottom();
+
+        return Math.max(1, (int) Math.ceil((float) totalHeightNoPadding / itemHeight));
+    }
+
+    @Override
+    public void onCoinMarketsSuccess(final CoinMarket[] coinMarkets) {
+        mMarketCoinAdapter.setCoinMarkets(Arrays.asList(coinMarkets));
+        disableShimmerEffect();
+    }
+
+    @Override
+    public void onCoinMarketsFail() {
+        Log.e(TAG, "Failed to load coin market list.");
+        disableShimmerEffect();
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (mShimmerActive) {
+            mShimmerViewContainer.startShimmer();
+        }
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        if (mShimmerActive) {
+            mShimmerViewContainer.stopShimmer();
+        }
+    }
+
+    private void disableShimmerEffect() {
+        if (mShimmerActive) {
+            mShimmerViewContainer.stopShimmer();
+            mShimmerViewContainer.setVisibility(View.GONE);
```

Comment:
nit: `mShimmerViewContainer.hideShimmer()`

Reply:
seems like we do need to hide the visibility as `hideShimmer` won't hide the added skeleton children.